File: 3. DSA/04. Python/03. Functions/Question02.py
# Leap Year
import math
class Solution:
    def isLeap (self, N):
        # code here
        if(N % 400 == 0 or (N % 100 != 0 and N % 4 == 0)):
            return 1
            
        else: 
            return 0

if __name__ == '__main__':
    year = int(input("Enter the year: "))

    obj = Solution()

    if(obj.isLeap(year)):
        print(f"\n{year} is a leap year")
    else:
        print(f"\n{year} is not a leap year")

File: 3. DSA/02. C++/03. Functions/Question02.cpp
// Leap Year

#include<bits/stdc++.h> 
using namespace std;

class Solution
{
public:
    int isLeap(int N)
    {
        //code here
        if(N % 400 == 0)
            return 1; // Leap Year
            
        else if(N % 4 == 0 && N % 100 != 0)
            return 1; // Leap Year
        
        else 
            return 0; // Normal Year
    }
}obj;

int main() 
{ 
    int year;
    cout << "Enter Year: ";
    cin >> year;

    // Solution obj;
    
    if(obj.isLeap(year))
        cout << "\nYear " << year << " is a leap year";
    else 
        cout << "\nYear " << year << " is not a leap year";

    return 0; 
}